cirq-core/cirq/transformers/analytical_decompositions/single_to_two_qubit_isometry_test.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import pytest
+import cirq
+
+VALID_INITIAL_STATES = [
+    np.array([1, 0, 0, 0]),
+    np.array([0, 1, 0, 0]),
+    np.array([1 / np.sqrt(2), -1 / np.sqrt(2), 0, 0]),
+    *[np.concatenate([cirq.testing.random_superposition(2), np.zeros(2)]) for _ in range(3)],
+]
+
+INVALID_INITIAL_STATES = [
+    np.array([1 / np.sqrt(2), 0, 0, -1 / np.sqrt(2)]),
+    cirq.testing.random_superposition(4),
+]
+
+
+@pytest.mark.parametrize(
+    'initial_state, is_valid',
+    list(
+        zip(
+            VALID_INITIAL_STATES + INVALID_INITIAL_STATES,
+            [True] * len(VALID_INITIAL_STATES) + [False] * len(INVALID_INITIAL_STATES),
+        )
+    ),
+)
+@pytest.mark.parametrize('unitary_matrix', [cirq.testing.random_unitary(4) for _ in range(10)])
+@pytest.mark.parametrize('allow_partial_czs', [True, False])
```

Comment:
These will generate _160_ test cases, which seems excessive. Is there a subset we can reduce this to?

Reply:
Reduced the number of test cases to 60 instead.